devdoc/job_queue.md

```diff
@@ -0,0 +1,59 @@
+# `job_queue` requirements
+
+`job_queue` is a module that allows jobs to be executed asynchronously. It queues jobs when there are no requests and queues requests when there are no jobs. When a job arrives and there are queued requests, it executes the least recently queued request with the job. When a request arrives and there are queued jobs, it executes the request with the least recently queued job.
+
+Consumers of `job_queue` must create a "`session`" before they can queue requests. A session can make two kinds of requests:
```

Comment:
>must create a "`session`" [](http://example.com/codeflow?start=25&length=25)

Why do we need a session for consumers? Or, what really **is** a session?

Reply:
A session is a consumer. `job_queue` guarantees that the requests submitted by a session will be completed in FIFO order of submission. The session identifies the consumer.